File: API/ContainerNinja.Contracts/DTO/CookedRecipeCalledIngredientDetailsDTO.cs
using ContainerNinja.Contracts.Data.Entities;
using ContainerNinja.Contracts.Enum;

namespace ContainerNinja.Contracts.DTO
{
    public class CookedRecipeCalledIngredientDetailsDTO
    {
        public int Id { get; set; }
        public int CookedRecipeId { get; set; }
        public CalledIngredientDTO? CalledIngredient { get; set; }
        public ProductStockDTO? ProductStock { get; set; }
        public int? ProductStockId { get; set; }
        public string Name { get; set; }
        public UnitType UnitType { get; set; }
        public float? Units { get; set; }

        public List<ProductStockDTO> ProductStockSearchItems { get; set; }
    }
}

File: Client/src/app/models/CookedRecipeCalledIngredientDetailsDTO.ts
import { CalledIngredientDTO } from "./CalledIngredientDTO";
import { ProductStockDTO } from "./ProductStockDTO";
import { UnitTypeDTO } from "./UnitTypeDTO";

export class CookedRecipeCalledIngredientDetailsDTO {
  id?: number = undefined;
  cookedRecipeId?: number = undefined;
  calledIngredient?: CalledIngredientDTO = undefined;
  productStock?: ProductStockDTO = undefined;
  productStockId?: number = undefined;
  name?: string = undefined;
  unitType?: UnitTypeDTO = undefined;
  units?: number = undefined;
  productStockSearchItems?: ProductStockDTO[] = undefined;
}
